core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -289,19 +288,20 @@ public void registerSuggester(String key, Suggester<?> suggester) {
      * <ul>
      * <li>Register the {@link ScoreFunctionParser} which parses XContent into a {@link ScoreFunctionBuilder} using its {@link ParseField}
      * </li>
-     * <li>Register the {@link Writeable.Reader} which reads a stream representation of the builder under a single name. This name must be
-     * the same name as the builder returns for getWriteableName.</li>
+     * <li>Register the {@link Writeable.Reader} which reads a stream representation of the builder under the
+     * {@linkplain ParseField#getPreferredName()}.</li>
      * </ul>
      */
-    public <T extends ScoreFunctionBuilder<T>> void registerScoreFunction(ParseField parserName, ScoreFunctionParser<T> parser,
-            String writeableName, Writeable.Reader<T> reader) {
```

Comment:
I hadn't see writeableName in here, now I get your previous comment on leaving it :)  but we had already agreed that getPrefferedName was ok with Christoph ;)